AlassaneWone/RoomMapping
# US-13 : Implementing Map Editing Features

**Description:**

This pull request finalizes the implementation of map editing features for the Drawing App. Users can now seamlessly modify their maps through a dedicated editing page accessible from the map list. Here are the key updates:

1. **Editing Workflow:**
   - Clicking the "Edit" button on a specific map in the map list now navigates the user to the dedicated map editing page.
   - The editing page presents a set of tools on the left and a canvas on the right, displaying the current state of the map.
   - Users can draw, add shapes, and place text zones on the canvas.

2. **Save Functionality:**
   - A "Save" button is now available after making modifications to the map.
   - Clicking "Save" records the edited map in Firestore, ensuring the changes persist.

3. **Toolset:**
   - Various tools are provided, including drawing, shape placement, and text zone creation.
   - Users can customize the color of shapes and text, providing a more personalized touch to their maps.

4. **User Interface:**
   - The UI is designed for an intuitive editing experience, ensuring easy navigation and tool accessibility.

**How to Test:**
    1. Navigate to the map list.
    2. Click "Edit" on a specific map to enter the editing page.
    3. Utilize the drawing tools, add shapes, and experiment with text zones.
    4. Click the "Save" button to ensure the changes are successfully saved in Firestore.

**Screenshots:**  

![Edit button](https://github.com/AlassaneWone/RoomMapping/assets/115146900/5a4d91c2-c33a-4365-9cff-92bc65e2ac24)  

![Editing example](https://github.com/AlassaneWone/RoomMapping/assets/115146900/d8e795cc-715b-4fe1-afcb-9286d50282c9)

**Checklist:**
- [ ] Map editing tools implemented successfully.
- [ ] Save functionality tested and operational.
- [ ] User interface design reviewed.
---

*Please review and provide feedback. Thank you!*

**US-13 : Implementing Map Editing Features**
